internal/controller/stas/containerimagescan_controller.go

```diff
@@ -79,10 +73,11 @@ func (r *ContainerImageScanReconciler) SetupWithManager(mgr ctrl.Manager) error
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&stasv1alpha1.ContainerImageScan{},
 			builder.WithPredicates(
-				predicate.Or(predicate.GenerationChangedPredicate{}, cisRescanDue(r.ScanInterval)),
+				predicate.GenerationChangedPredicate{},
 				ignoreDeletionPredicate(),
 			)).
 		WithEventFilter(predicate.And(predicates...)).
+		Watches(&source.Channel{Source: r.EventChan}, &handler.EnqueueRequestForObject{}).
```

Comment:
Shouldn't we enqueue the request to the owner here so that it can delete old CIS report and trigger a rescan?

Reply:
We try to avoid imperative operations. If we delete the CIS (as trivy-operator does) we will lose the last scan result. IMO it is better to have a scan report, even if not fully up-to-date.

Reply:
yeah we discussed this before, agree with that